app/controllers/bras_controller.rb

```diff
@@ -38,6 +38,23 @@ def edit
   end
 
   def update
+    # This @non_tags variable is somewhat confusing for me, partially because
+    # it's a variable named in the negative ('non_tags'), but also because it's
+    # tied into your ability model as an instance variable, which increases the
+    # implicit coupling between your controller and views (by that I mean you
+    # have to rememeber to set this variable as true or false in every action
+    # that renders a view that uses can? :update @bra)
+
+    # Additonally, I'm fairly certain there's a permissions bug here that would
+    # allow any user to update a bra they didn't create by injecting values
+    # into params for this action. This is a subtle bug, but in short, I'd
+    # suggest creating another action just for adding tags, and moving the
+    # logic for creating tags into that controller action. I don't often
+    # advocate for that, but in this case, you can handle authorization for
+    # bra#update and tag#create completely independently. And the
+    # tags_controller#create action can be more limited than this update action,
+    # which allows updating of any bra attribute due to the un-limited
+    # `@bra.update(bra_params)` line below.
```

Comment:
Good tip. This was the most challenging part for me, and it was also the last thing I added. If I do decide to take the project further I will definitely fix it!